runtime/include/chpl-mem-array.h

```diff
@@ -57,7 +57,7 @@ void* chpl_mem_array_alloc(size_t nmemb, size_t eltSize,
                            int32_t lineno, int32_t filename) {
   void* p = NULL;
   const size_t size = nmemb * eltSize;
-#ifdef HAS_GPU_LOCALE
+#if defined(HAS_GPU_LOCALE) && !defined(GPU_RUNTIME_CPU)
```

Comment:
> Though that would add the risk of a super confusing situation where it might not be defined where you need it to be if that common header. I'm not going to push on it, but something to consider.

This scared me. I'll leave it as-is.